projects/storefrontlib/src/cms-components/user/login/login.component.ts

```diff
@@ -1,33 +1,32 @@
-import { Component } from '@angular/core';
-import { AuthService, User, UserService } from '@spartacus/core';
-import { Observable } from 'rxjs';
-import { filter, map, switchMap } from 'rxjs/operators';
-import { LoginComponentService } from './login.component.service';
+import { ChangeDetectorRef, Component, OnInit, OnDestroy } from '@angular/core';
+import { User, UserService } from '@spartacus/core';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'cx-login',
   templateUrl: './login.component.html',
 })
-export class LoginComponent {
+export class LoginComponent implements OnInit, OnDestroy {
+  userDetails: User = {};
+  private subscription = new Subscription();
+
   constructor(
-    private auth: AuthService,
     private userService: UserService,
-    private loginService: LoginComponentService
+    private cd: ChangeDetectorRef
   ) {}
 
-  get user$(): Observable<User> {
-    return this.auth.getUserToken().pipe(
-      map(token => {
-        if (token && !!token.access_token && !this.loginService.isLogin) {
-          this.loginService.isLogin = true;
-          this.userService.load();
-        } else if (token && !token.access_token && this.loginService.isLogin) {
-          this.loginService.isLogin = false;
-        }
-        return token;
-      }),
-      filter(token => token && !!token.access_token),
-      switchMap(() => this.userService.get())
+  ngOnInit(): void {
```

Comment:
No need subscription. We only need this:
`user$ = this.userService.get();`
And in the html, use `async` pipe.

Reply:
It became a bit messy in the template when I did this.
Is there a simple way to handle the logic in the template if we use the async pipe variant?